### PR TITLE
Added options, features for asset-swapper

### PR DESCRIPTION
**Pull request by dave4506**
## Description

Added the following: 
- Option to disable fee order logic
- Option to default swap quote consumer to a certain `ConsumerType`
- Added a utils library for developers to check allowances for taker
- Optimized `SwapQuoteConsumer` output (deduped data for lower gas consumption)

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/asset-swapper/src/types.ts [hunk 1]
```diff
@@ -262,6 +272,7 @@ export interface SwapQuoteInfo {
  */
```
Comment (fragosti) [inline packages/asset-swapper/src/types.ts]: You forgot to add doc strings.

File: packages/asset-swapper/src/types.ts [hunk 2]
```diff
@@ -198,6 +207,7 @@ export type SwapQuote = MarketBuySwapQuote | MarketSellSwapQuote;
...
```
Comment (fragosti) [inline packages/asset-swapper/src/types.ts]: Are these docstrings correct?

File: packages/asset-swapper/src/utils/swap_quote_utils.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+        const provider = providerUtils.standardizeOrThrow(supportedProvider);
+        this.provider = provider;
+        this.networkId = networkId;
+        this._contractWrappers = new ContractWrappers(this.provider, {
+            networkId,
+        });
+    }
+
+    public async isTakerAddressAllowanceEnoughForBestAndWorstQuoteInfoAsync(
```
Comment (fragosti) [inline packages/asset-swapper/src/utils/swap_quote_utils.ts]: Should we remove this class and just put this util on `SwapQuoter`?

File: packages/asset-swapper/test/swap_quote_calculator_test.ts [hunk 4]
```diff
@@ -78,6 +78,7 @@ describe('swapQuoteCalculator', () => {
                         new BigNumber(200),
                         0,
                         false,
+                        false,
```
Comment (fragosti) [inline packages/asset-swapper/test/swap_quote_calculator_test.ts]: Tests for the true condition would be nice.

Comment (dave4506) [reply packages/asset-swapper/test/swap_quote_calculator_test.ts]: Yeah, will be adding coverage in a seperate PR

**Pull request merged**